plugins/asus-hid/fu-asus-hid-device.c

```diff
@@ -515,19 +525,35 @@ fu_asus_hid_device_set_quirk_kv(FuDevice *device,
 	return FALSE;
 }
 
+static void
+fu_asus_hid_device_finalize(GObject *object)
+{
+	FuAsusHidDevice *self = FU_ASUS_HID_DEVICE(object);
+
+	g_signal_handler_disconnect(FU_DEVICE(object), self->child_added_id);
```

Comment:
maybe move to destroy? just remember the != 0 and setting it to 0 if set.

Reply:
I'm not sure I really follow, Like a `GDestroyNotify` on a signal?

Reply:
I pushed a fixup, hope that's okay.

Reply:
Sure! ah you meant dispose.  Thanks will squash it.